gitversioner/src/main/kotlin/com/pascalwelsch/gitversioner/GitVersioner.kt

```diff
@@ -190,22 +193,31 @@ public open class GitVersioner internal constructor(
                         DEFAULT_SHORT_NAME_FORMATTER(versioner).toString()
                     }
 
-                    sb.append("-").append(shortName)
+                    sb.append("-$shortName")
                 }
 
                 val featureCount = featureBranchCommits.count()
-                if (featureCount > 0) {
-                    sb.append("+").append(featureCount)
+                if (featureCount > 0 && addFeatureBranchCommitCount) {
+                    sb.append("+$featureCount")
                 }
                 if (localChanges != NO_CHANGES) {
                     if (addSnapshot) {
                         sb.append("-SNAPSHOT")
                     }
                     if (addLocalChangesDetails) {
-                        sb.append("(").append(localChanges).append(")")
+                        sb.append("($localChanges)")
+                    }
+                }
+                if (isFeatureBranch && addTimestamp) {
+                    sb.append("-${System.currentTimeMillis() / 1000}")
```

Comment:
using the system time is not reproducable. What about the latest commit date?

Reply:
Uniqueness was the goal. Build systems/IDEs routinely cache packages. Using anything that doesn't change by itself entails committing for every build - a suboptimal situation. Plus, snapshots of feature branches should _not_ guarantee reproducibility.

As a compromise, a timestamp could be applied only if there are uncommitted local changes?

Reply:
For background, all of this arouse out of cross-compiling Kotlin to Java and JS both, then using NPM over JS. Unlike Maven, NPM doesn't have a usable way to publish to local, so artifacts have to be round-tripped to the repo, which makes every build effectively public.